### ISSUE TITLE
modopt doc id generation is broken

### ISSUE THREAD

**Issue by twsouthwick** [Microsoft/dotnet-apiport#236]
One test is failing because it expects:
`M:TestClass.Bar(System.SByte optmod System.Runtime.CompilerServices.IsConst reqmod System.Runtime.CompilerServices.IsSignUnspecifiedByte*)`
but the only close to it in the actual results is:
`M:TestClass.Bar(System.SByte reqmod System.Runtime.CompilerServices.IsSignUnspecifiedByte optmod System.Runtime.CompilerServices.IsConst*)`

Original IL: [modopt.il](https://github.com/Microsoft/dotnet-apiport/blob/master/tests/Microsoft.Fx.Portability.MetadataReader.Tests/Tests/modopt.il#L23)

Is it possible the expected result is wrong?
Reflector shows: 
 `public unsafe void Bar(sbyte modreq(IsSignUnspecifiedByte) modopt(IsConst)* x); `
 CCI shows: 
`System.Void TestClass.Bar(System.SByte optmod System.Runtime.CompilerServices.IsConst reqmod System.Runtime.CompilerServices.IsSignUnspecifiedByte*)`

Maybe its a bug in CCI?

Seen when upgrading from System.Reflection.Metadata 1.2.0-alpha-00017 to 1.2.0-rc2-23713 with PR #232

**Comment by twsouthwick** [comment]
@nguerrera does this look like a S.R.Metadata issue to you or something in our code?

**Comment by nguerrera** [comment]
This does not look like a S.R.Metadata issue. The IL syntax (like what is shown in the reflector case is uses the reverse of the metadata order. i.e. `A modreq(B) modopt(C)` is `CMOD_OPT C CMOD_REQD B A` in the metadata signature. Think of the decoder as building this type as tree:

```
GetModifiedType(required: false, C,
    GetModifiedType(required: true, B,
          A);
```

How you serialize that tree to a string is up to you and whether you're bound by the semantics of a particular syntax. It looks like that CCI syntax prints the modifiers out in metadata order rather than IL syntactic order.

I wrote some test code that spread previously that printed out modifiers in IL syntax but metadata order. That was wrong and possibly the source of the change if you copied it.

**Comment by twsouthwick** [comment]
Thanks @nguerrera for the explanation. @conniey I've fixed this in #239
